File: 203-remove-linked-list-elements/203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if (head == null){
            return null;
        }
        ListNode nwlist= null;
        while (head != null){
            if(head.val != val){
                nwlist = new ListNode(head.val, nwlist);
            }
            head = head.next;
        }
        head = null;
        while (nwlist != null){
            head = new ListNode(nwlist.val, head);
            nwlist = nwlist.next;
        }
        return head;
    }
}

File: 203-remove-linked-list-elements/203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy = ListNode(-1)
        dummy.next = head
        start, end = dummy, dummy
        curr = head
        
        while curr:
            if curr.val != val:
                end.next = curr
                end = end.next
            
            curr = curr.next
        
        end.next = None
        return start.next
        